lib/project_types/rails/commands/create.rb

```diff
@@ -65,12 +68,51 @@ def self.help
 
       private
 
+      def check_node
+        cmd_path = @ctx.which('node')
+        if cmd_path.nil?
+          @ctx.abort(@ctx.message('rails.create.error.node_required')) unless @ctx.windows?
+          @ctx.puts("{{x}} {{red:" + @ctx.message('rails.create.error.node_required') + "}}")
+          @ctx.puts(@ctx.message('rails.create.info.open_new_shell', 'node'))
+          raise ShopifyCli::AbortSilent
+        end
+
+        version, stat = @ctx.capture2e('node', '-v')
+        unless stat.success?
+          @ctx.abort(@ctx.message('rails.create.error.node_version_failure')) unless @ctx.windows?
+          # execution stops above if not Windows
+          @ctx.puts("{{x}} {{red:" + @ctx.message('rails.create.error.node_version_failure') + "}}")
+          @ctx.puts(@ctx.message('rails.create.info.open_new_shell', 'node'))
+          raise ShopifyCli::AbortSilent
+        end
+
+        @ctx.done(@ctx.message('rails.create.node_version', version))
+      end
+
+      def check_yarn
+        cmd_path = @ctx.which('yarn')
+        if cmd_path.nil?
+          @ctx.abort(@ctx.message('rails.create.error.yarn_required')) unless @ctx.windows?
+          @ctx.puts("{{x}} {{red:" + @ctx.message('rails.create.error.yarn_required') + "}}")
+          @ctx.puts(@ctx.message('rails.create.info.open_new_shell', 'yarn'))
+          raise ShopifyCli::AbortSilent
+        end
+
+        version, stat = @ctx.capture2e('yarn', '-v')
+        unless stat.success?
+          @ctx.abort(@ctx.message('rails.create.error.yarn_version_failure')) unless @ctx.windows?
+          @ctx.puts("{{x}} {{red:" + @ctx.message('rails.create.error.yarn_version_failure') + "}}")
+          @ctx.puts(@ctx.message('rails.create.info.open_new_shell', 'yarn'))
+          raise ShopifyCli::AbortSilent
+        end
+
+        @ctx.done(@ctx.message('rails.create.yarn_version', version))
+      end
+
       def build(name, db)
-        install_gem('rails')
-        CLI::UI::Frame.open(@ctx.message('rails.create.installing_bundler')) do
-          install_gem('bundler', '~>1.0')
+        @ctx.abort(@ctx.message('rails.create.error.install_failure', 'rails')) unless install_gem('rails')
+        @ctx.abort(@ctx.message('rails.create.error.install_failure', 'bundler ~>2.0')) unless
```

Comment:
Just checking - we seem to be dropping the frame here when installing the gems, any particular reason for it?

Reply:
The previous code wasn't working anyway, and `bundler 2.x` is preferable.  Didn't see necessary to have a single installation within its own frame, so simplified it.